xm-commons-logging-configurable/src/main/java/com/icthh/xm/commons/logging/configurable/LoggingRefreshableConfiguration.java

```diff
@@ -0,0 +1,136 @@
+package com.icthh.xm.commons.logging.configurable;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import com.icthh.xm.commons.logging.config.LoggingConfig;
+import com.icthh.xm.commons.logging.config.LoggingConfig.LepLogConfiguration;
+import com.icthh.xm.commons.logging.config.LoggingConfig.LogConfiguration;
+import com.icthh.xm.commons.logging.config.LoggingConfigService;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.MapUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.icthh.xm.commons.tenant.TenantContextUtils.getRequiredTenantKeyValue;
+
+@Slf4j
+@Component
+@Primary
+public class LoggingRefreshableConfiguration implements RefreshableConfiguration, LoggingConfigService {
+
+    private final ConcurrentHashMap<String, Map<String, LogConfiguration>> serviceLoggingConfig = new ConcurrentHashMap<>();
```

Comment:
User Map interface as the variable type.

Reply:
changed